plugins/generator-1.20.4/neoforge-1.20.4/templates/elementinits/potioneffects.java.ftl

```diff
@@ -29,21 +29,58 @@
 -->
 
 <#-- @formatter:off -->
+<#include "../procedures.java.ftl">
 
 /*
  *	MCreator note: This file will be REGENERATED on each build.
  */
 
 package ${package}.init;
 
-public class ${JavaModName}MobEffects {
+<#assign effects_that_expire = potioneffects?filter(effect -> hasProcedure(effect.onExpired))>
 
-	public static final DeferredRegister<MobEffect> REGISTRY = DeferredRegister.create(ForgeRegistries.MOB_EFFECTS, ${JavaModName}.MODID);
+<#if effects_that_expire?size != 0>@Mod.EventBusSubscriber </#if>public class ${JavaModName}MobEffects {
+
+	public static final DeferredRegister<MobEffect> REGISTRY = DeferredRegister.create(Registries.MOB_EFFECT, ${JavaModName}.MODID);
 
 	<#list potioneffects as effect>
-	public static final RegistryObject<MobEffect> ${effect.getModElement().getRegistryNameUpper()} =
+	public static final DeferredHolder<MobEffect, ${effect.getModElement().getName()}MobEffect> ${effect.getModElement().getRegistryNameUpper()} =

```

Comment:
Do we need `<MobEffect, InstanceClassName>`, or is `<MobEffect, MobEffect>` enough (shorter code, one FTL replacement less)?

Reply:
Not strictly needed